isisd/isis_tlvs.c

```diff
@@ -2789,6 +2808,18 @@ static int unpack_tlv_router_cap(enum isis_tlv_context context,
 				stream_forward_getp(
 					s, length - SR_ALGORITHM_COUNT);
 			break;
+		case ISIS_SUBTLV_SRLB:
+			/* Ignore Flags which are not defined */
+			stream_getc(s);
+			rc->srlb.range_size = stream_get3(s);
```

Comment:
Ideally we need to handle the case where multiple SRLB ranges are being advertised (to prevent LSP parsing errors when that happens).

Reply:
RFC8667 section 3.3 mention that a router must advertise only one SRLB. But, I agree with your remarks and I add a verification to skip subsequent SRLB keeping only the first one as mention in the RFC8667.

I also added the same verification for SRGB while the code support for the moment only one block.

Reply:
That RFC says that a router shouldn't advertise more than one SRLB sub-TLV, but it can still advertise multiple SRLBs in the same SRLB sub-TLV. In that case we'd need to call `stream_forward_getp()` to ignore the remaining SRLB Descriptor entries. This is a non-blocking comment.

Reply:
I think you have not review the latest version of the code since I had this piece of code:
```
case ISIS_SUBTLV_SRLB:
	/* RFC 8667 section #3.3: Only one SRLB is authorized */
	if (rcap->srlb.range_size != 0) {
		stream_forward_getp(s, length);
		continue;
	}
```
At line 2817

Reply:
That code handles the case where you receive multiple SRLB sub-TLVs, but you can have multiple SRLBs in the same SRLB sub-TLV. This is a minor issue that can be fixed in future PRs.

Reply:
>That code handles the case where you receive multiple SRLB sub-TLVs, but you can have multiple SRLBs in the same SRLB sub-TLV. This is a minor issue that can be fixed in future PRs.

There may be several SRLB sub-TLV within the LSP, but no multiple SRLBs in the same SRLB sub-TLV. Look at RFC 8667:

>   The SRLB sub-TLV is used for this purpose and has following format:

    0                   1                   2                   3
    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   |   Type        |     Length    |    Flags      |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+

   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   |                     Range                     |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   //                SID/Label Sub-TLV (variable)                 //
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+

      Type:    22

      Length:  Variable

      Flags:   1 octet of flags.  None are defined at this stage.

      One or more SRLB Descriptor entries, each of which have the
      following format:

         Range:   3 octets

         SID/Label sub-TLV:  MPLS label as defined in Section 2.3

What is variable, like for the SRGB, is the encoding of SID/Label: 3 bytes if it is a label and 4 bytes if it is an index. But, SID/Label is unique per SRGB or SRLB. Indeed, a SID/Label is always associated to a range. Thus, even for SRGB, you could not have multiple SID/Label subTLV within a SRLB or SRGB. What you could collect is multiple SRGB or SRLB. Thus, the code take care of that and select the first SRLB as mention later in the same section of the RFC:

>    The SRLB sub-TLV MAY be advertised in an LSP of any number, but a
   router MUST NOT advertise more than one SRLB sub-TLV.  A router
   receiving multiple SRLB sub-TLVs, from the same originator, SHOULD
   select the first advertisement in the lowest-numbered LSP.

What you propose in term of code is to protect parsing again malformed subTLVs. I agree that we could add this piece of code. But, we must do it for all TLVs, subTLVs ... and try to envisage all possible corruption. So, If we would be sure, we must check the expected size regarding the type of subTLV. So, for SRLB, we must check that the size is 9 bytes (label) or 10 bytes (index) and skip subTLV if it not corresponds to that sizes. I'll add this test as well as for SRGB.